lib/types/HttpStream.ts

```diff
@@ -42,5 +43,45 @@ export class HttpStream {
   ) {
     this.stream = readableStream;
     this.totalBytes = totalBytes;
+    this.destroyed = readableStream.destroyed;
+  }
+
+  /**
+   * Returns if the stream is errored
+   */
+  isErrored (): boolean {
```

Comment:
Nitpicking / coding style : you could use a getter named `errored` here for example